src/App.tsx

```diff
@@ -1,9 +1,87 @@
+import { Route, Routes, useLocation } from 'react-router-dom';
 import './App.scss';
+import { useEffect } from 'react';
+// import { Product } from './Components/ProductCards/Product';
+import { FavoriteContextProvider } from './FavoriteContext';
+import { HomePAge } from './Components/Pages/HomePage/HomePage';
+import { Header } from './Components/Header/Header';
+import { NotFoundPage } from './Components/Pages/NotFoundPage/NotFoundPage';
+import { PhonesPage } from './Components/Pages/PhonesPage/PhonesPage';
+import { PhonePreview } from './Components/Pages/PhonePreview';
+import { ContactsPage } from './Components/Pages/Contacts/Contacts';
+import { Favorite } from './Components/Pages/Favorite/Favorite';
+import { Basket } from './Components/Pages/Basket/Basket';
+import { Rights } from './Components/Pages/Rights/Rights';
+import { Footer } from './Components/Footer/Footer';
+// import { BlockBasket } from './Components/Pages/Basket/BlockBasket';
 
-const App = () => (
-  <div className="App">
-    <h1>React Phone Catalog</h1>
-  </div>
-);
+export const App = () => {
+  const location = useLocation();
 
-export default App;
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [location]);
+
+  return (
+    <>
+      <FavoriteContextProvider>
+        <Header searchValue="" setSearchValue={() => null} />
+
+        <div className="main">
+          <Routes>
+            <Route path="/" element={<HomePAge />} />
+            <Route path="/phones" element={<PhonesPage />} />
+            <Route path="/phones/:id" element={<PhonePreview />} />
+            <Route
+              path="/tablets"
+              element={(
+                <NotFoundPage
+                  title="Tables"
+                  h1="Tablet page under construction"
```

Comment:
would be better to rename this `h1` to `header` for example

Reply:
I understand, but I did it on purpose so as not to get confused, since this h1 is used at all times when it says that the page was not found, I'll figure out what to do with it